inc/wp-forms-api.js

```diff
@@ -161,8 +161,14 @@
 				view.model.fetch();
 
 				// Don't save input name as part of the model as it should be invariant
-				view.input_name = $(this).find('input').attr('name');
-				view.input_type = $(this).data('attachment-type');
+				if(typeof this.name === 'undefined'){
+					// New widgets added to sidebar after page load
```

Comment:
Hmmmm, so this doesn't necessarily relate to widgets: this is a generic routine that exists outside of the specific case that is being fixed here.

I would rather see this written as:

`view.model.set('input_name', this.name || $(this).find('input').attr('name'));`

Unless you're changing more than one model property, the `model.set('prop', 'value')` form is preferred.